agent-identity/python/on-behalf-of-flow/sleeping-guardian/aurelius_agent_llm.py

```
@@ -0,0 +1,490 @@
+        except ToolException as e:
+            if INSUFFICIENT_SCOPE_SENTINEL in str(e):
+                print(f"\n{'='*80}")
+                print(f"[Aurelius AI] 🔐 INSUFFICIENT SCOPE ERROR DETECTED")
+                print(f"[Aurelius AI] Error: {str(e)}")
+                print(f"[Aurelius AI] Current Scopes: {self.current_scopes}")
+                print(f"[Aurelius AI] Triggering CIBA authorization...")
+                print(f"{'='*80}\n")
+                await self._handle_insufficient_scope(local_state, str(e))
+            else:
+                print(f"[Aurelius AI] ⚠️  Tool error: {e}")
+        except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)